glommio/src/io/bulk_io.rs

```diff
@@ -72,240 +32,529 @@ impl IoVec for (u64, usize) {
     }
 }
 
-#[derive(Debug, Copy, Clone)]
-pub(crate) struct ReadManyArgs<V: IoVec + Unpin> {
-    pub(crate) user_read: V,
-    pub(crate) system_read: (u64, usize),
-}
+pub(crate) struct MergedIOVecs<V: IoVec>(pub(crate) VecDeque<V>, pub(crate) (u64, usize));
 
-/// A stream of ReadResult produced asynchronously.
-///
-/// See [`DmaFile::read_many`] for more information
-#[derive(Debug)]
-pub struct ReadManyResult<V: IoVec + Unpin> {
-    pub(crate) inner: OrderedBulkIo<ReadManyArgs<V>>,
-    pub(crate) current: Option<ScheduledSource>,
+impl<V: IoVec> Unpin for MergedIOVecs<V> {}
```

Comment:
Why is this needed? I thought it would be automatically implemented.

Reply:
`Unpin` is hard so I may be doing something wrong here. If I remove that I won't be able to compile until I require the user to pass an `Unpin` `IoVec` which I think is unnecessary (stream never require that from what I can see).

Open to suggestion on this one.

Reply:
`IoVec` is `impl`'d by glommio, not by users, right? It seems like adding the `Unpin` bound to the returned `IoVec` type in `read_may` should be fine.

I also think these `impl Unpin` are correct, I was just surprised because I hadn't come across the need for them yet.

No objections on either direction you want to go.

Reply:
No, `IoVec` is a trait. Making it one makes the user code a lot more flexible because users can give us structs with a lot more context that's useful when the IO comes back. 

That being said, maybe there is no way around requiring them to implement `Unpin` themselves? I was trying to avoid that.

Reply:
I dropped 7e73bf1f721659d70c3389058c188b849ab95a74 so the `IoVec` the user gives us need to implement `Unpin`.